src/Razor/Microsoft.NET.Sdk.Razor/src/ConcatenateCssFiles.cs

```diff
@@ -13,21 +14,72 @@ namespace Microsoft.AspNetCore.Razor.Tasks
-                foreach (var line in File.ReadLines(FilesToProcess[i].GetMetadata("FullPath")))
+                // We are importing bundles from other class libraries and packages, in that case we need to compute the
+                // import path relative to the position of where the final bundle will be.
+                // Our final bundle will always be at "<<CurrentBasePath>>/scoped.styles.css"
+                // Other bundles will be at "<<BundleBasePath>>/bundle.bdl.scp.css"
+                // The base and relative paths can be modified by the user, so we do a normalization process to ensure they
+                // are in the shape we expect them before we use them.
+                // We normalize path separators to '\' from '/' which is what we expect on a url. The separator can come as
+                // '\' as a result of user input or another MSBuild path normalization operation. We always want '/' since that
+                // is what is valid on the url.
+                // We remove leading and trailing '/' on all paths to ensure we can combine them properly. Users might specify their
+                // base path with or without forward and trailing slashes and we always need to make sure we combine them appropriately.
+                // These links need to be relative to the final bundle to be independent of the path where the main app is being served.
+                // For example:
+                // An app is served from the "subdir" path base, the main bundle path on disk is "MyApp/scoped.styles.css" and it uses a
+                // library with scoped components that is placed on "_content/library/bundle.bdl.scp.css".
+                // The resulting import would be "import '../_content/library/bundle.bdl.scp.css'".
+                // If we were to produce "/_content/library/bundle.bdl.scp.css" it would fail to accoutn for "subdir"
+                // We could produce shorter paths if we detected common segments between the final bundle base path and the imported bundle
+                // base paths, but its more work and it will not have a significant impact on the bundle size size.
+                var normalizedBasePath = NormalizePath(ScopedCssBundleBasePath);
+                var currentBasePathSegments = normalizedBasePath.Split(new[] { '/' }, StringSplitOptions.RemoveEmptyEntries);
+                var prefix = string.Join("/", Enumerable.Repeat("..", currentBasePathSegments.Length));
+                for (var i = 0; i < ProjectBundles.Length; i++)
+                {
+                    var bundle = ProjectBundles[i];
+                    var bundleBasePath = NormalizePath(bundle.GetMetadata("BasePath"));
+                    var relativePath = NormalizePath(bundle.GetMetadata("RelativePath"));
+                    var importPath = NormalizePath(Path.Combine(prefix, bundleBasePath, relativePath));
+
+                    builder.AppendLine($"import '{importPath}'");
```

Comment:
Isn't the syntax `@import`, not `import`?

Just want to check we've definitely got coverage to say this works for real. If there's a possibility of adding an E2E test that uses `getComputedStyle` to verify that a external-package-provided scoped style has actually been applied, that would be ideal. If that's too much to add at this stage, let's have an issue tracking the need for E2E coverage and make sure we do a fair amount of manual verification.

Reply:
Yeah, I don't know how I missed that. I'll change it.

Reply:
Cool. Let's not miss the E2E test need though too, to check the style truly appears in the browser.

Reply:
BTW there's already an E2E test that verifies styles from an RCL get applied properly: see `CanUseComponentAndStaticContentFromExternalNuGetPackage` in `Microsoft.AspNetCore.Components.E2ETest.Tests.ComponentRenderingTest`. It reads a computed style.

So it's probably not too hard to update how that test works so the style comes from a scoped CSS file.

Reply:
I added an E2E test for this, just took one of the styles in the TestContentPackage styles and put it on a scoped css file.